[distributed] Healing code crashed on packet cluster

```
Drive Capacity: 1.6 TiB Free, 1.7 TiB Total
Status:         16 Online, 0 Offline. We can withstand [8] more drive failure(s).
panic: runtime error: index out of range

goroutine 311577727 [running]:
panic(0xabd720, 0xc420010140)
        /home/minio/go/src/runtime/panic.go:500 +0x1a1
github.com/minio/minio/cmd.listDirHealFactory.func1(0xc44d3c9648, 0x8, 0xc4b91ad580, 0x31, 0x0, 0x0, 0x0, 0x18, 0xbe8c7d, 0xbe8c7c, ...)
        /home/minio/work/src/github.com/minio/minio/cmd/xl-v1-list-objects-heal.go:61 +0x5fd
github.com/minio/minio/cmd.doTreeWalk(0xc44d3c9648, 0x8, 0xc4b91ad580, 0x31, 0x0, 0x0, 0x0, 0x0, 0x1, 0xc4aa058f90, ...)
        /home/minio/work/src/github.com/minio/minio/cmd/tree-walk.go:163 +0xe9
github.com/minio/minio/cmd.doTreeWalk(0xc44d3c9648, 0x8, 0xc468c6d980, 0x2c, 0x0, 0x0, 0x0, 0x0, 0x1, 0xc4aa058f90, ...)
        /home/minio/work/src/github.com/minio/minio/cmd/tree-walk.go:220 +0x432
github.com/minio/minio/cmd.doTreeWalk(0xc44d3c9648, 0x8, 0xc43cf58690, 0x28, 0x0, 0x0, 0x0, 0x0, 0x1, 0xc4aa058f90, ...)
        /home/minio/work/src/github.com/minio/minio/cmd/tree-walk.go:220 +0x432
github.com/minio/minio/cmd.doTreeWalk(0xc44d3c9648, 0x8, 0xc472f31110, 0x26, 0x0, 0x0, 0x0, 0x0, 0x1, 0xc4aa058f90, ...)
        /home/minio/work/src/github.com/minio/minio/cmd/tree-walk.go:220 +0x432
github.com/minio/minio/cmd.doTreeWalk(0xc44d3c9648, 0x8, 0xc4bff535c0, 0x1b, 0x0, 0x0, 0x0, 0x0, 0x1, 0xc4aa058f90, ...)
        /home/minio/work/src/github.com/minio/minio/cmd/tree-walk.go:220 +0x432
github.com/minio/minio/cmd.doTreeWalk(0xc44d3c9648, 0x8, 0xc4ca5eb7a0, 0x14, 0x0, 0x0, 0x0, 0x0, 0x1, 0xc4aa058f90, ...)
        /home/minio/work/src/github.com/minio/minio/cmd/tree-walk.go:220 +0x432
github.com/minio/minio/cmd.doTreeWalk(0xc44d3c9648, 0x8, 0xc4215f8260, 0x11, 0x0, 0x0, 0x0, 0x0, 0x1, 0xc4aa058f90, ...)
        /home/minio/work/src/github.com/minio/minio/cmd/tree-walk.go:220 +0x432
github.com/minio/minio/cmd.doTreeWalk(0xc44d3c9648, 0x8, 0xc4e001f830, 0xb, 0x0, 0x0, 0x0, 0x0, 0x1, 0xc4aa058f90, ...)
        /home/minio/work/src/github.com/minio/minio/cmd/tree-walk.go:220 +0x432
github.com/minio/minio/cmd.doTreeWalk(0xc44d3c9648, 0x8, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0xc4202f8701, 0xc4aa058f90, ...)
        /home/minio/work/src/github.com/minio/minio/cmd/tree-walk.go:220 +0x432
github.com/minio/minio/cmd.startTreeWalk.func1(0xc44d3c9648, 0x8, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0xc4be2b7101, 0xc4aa058f90, ...)
        /home/minio/work/src/github.com/minio/minio/cmd/tree-walk.go:261 +0xe0
created by github.com/minio/minio/cmd.startTreeWalk
```

--- a/cmd/xl-v1-list-objects-heal.go
+++ b/cmd/xl-v1-list-objects-heal.go
@@ -58,7 +58,9 @@ func listDirHealFactory(isLeaf isLeafFunc, disks ...StorageAPI) listDirFunc {
 			// find elements in entries which are not in mergedentries
 			for _, entry := range entries {
 				idx := sort.SearchStrings(mergedEntries, entry)
-				if mergedEntries[idx] == entry {
+				// idx different from len(mergedEntries) means entry is not found
+				// in mergedEntries
+				if idx < len(mergedEntries) {
 					continue
 				}
 				newEntries = append(newEntries, entry)

